### DISCUSSION TITLE
It possible to update theme in real time?

### DISCUSSION THREAD

**Discussion by jeffreyepstein777** [TomSchimansky/CustomTkinter#2373] [Q&A]
Its only apply to new window after the change of theme
`script_dir = os.path.dirname(os.path.abspath(__file__))
#Path at the folder of main.py
NightTrain = os.path.join(script_dir, "NightTrain.json")
current_theme = "NightTrain"
NeonBanana = os.path.join(script_dir, "NeonBanana.json")
customtkinter.set_ctk_parent_class(tkinterDnD.Tk)
customtkinter.deactivate_automatic_dpi_awareness()
customtkinter.set_appearance_mode("dark")  # Modes: "System" (standard), "Dark", "Light"
customtkinter.set_default_color_theme(NightTrain)  # Themes: "blue" (standard), "green", "dark-blue"

def change_theme():
    global current_theme  #Global theme

    if current_theme == "NightTrain":
        current_theme = "NeonBanana"
        customtkinter.set_default_color_theme(NeonBanana)
    elif current_theme == "NeonBanana":
        current_theme = "NightTrain"
        customtkinter.set_default_color_theme(NightTrain)
    else:
        print("unknown theme:", current_theme)

username1 = "user" + str(random.randint(0, 9999))

save_file = "data.json"`

**Comment by avalon60** [comment]
Hello @purplesmileguy. If I understand your question, it's a limitation of CustomTkinter. There is a lot of heavy lifting involved, in order to change themes, after an application has started. You would need to destroy and re-create all your widgets after changing themes.

This would require that you design your application around the ability, as you also need to take into account the session state (i.e. any user inputs made etc).

**Reply by rigvedmaanas** [reply]
Hi @avalon60, 

Maybe you don't need to destroy and recreate all the widgets. You could just loop over every widget in the window and configure the color from the json file. Probably just changing the color of the widget won't change the session state.

**Reply by avalon60** [reply]
Either way, it's still a lot of heavy lifting, and you'd also need to update all of the widget geometry, by doing it this way.

**Reply by JanPanthera** [reply]
You can create a wrapper class around a ctk Widget that has a method for updating, i use it for localization fore example. When i change the language or a yo a theme, you can send this event. And the wraper class recieves it and handles it. Offcourse it is some work but way easier then you can currently imagine. 😹

**Comment by avalon60** [comment]
> You can create a wrapper class around a ctk Widget that has a method for updating, i use it for localization fore example. When i change the language or a yo a theme, you can send this event. And the wraper class recieves it and handles it. Offcourse it is some work but way easier then you can currently imagine. 😹

You would surely need a whole bunch of wrapper classes; the widgets have a variety (varied compositions) of attributes. Also, some of the widgets are compound widgets, made up of more than one CustomTkinter widget (e.g. CTkTabView, CTkScrollableFrame, CTkTextbox). These may add extra challenges (e.g issue #1803 - although this example may be fixed now).
